test/unit/controllers/declaration/SupervisingCustomsOfficeControllerSpec.scala

```diff
@@ -0,0 +1,199 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package unit.controllers.declaration
+
+import controllers.declaration.SupervisingCustomsOfficeController
+import helpers.views.declaration.WarehouseIdentificationMessages
+import models.{DeclarationType, Mode}
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito
+import org.mockito.Mockito.{verify, when}
+import org.scalatest.{BeforeAndAfterEach, OptionValues}
+import play.api.libs.json.Json
+import play.api.test.Helpers._
+import play.twirl.api.HtmlFormat
+import unit.base.ControllerSpec
+import views.html.declaration.supervising_customs_office
+
+class SupervisingCustomsOfficeControllerSpec extends ControllerSpec with BeforeAndAfterEach with WarehouseIdentificationMessages with OptionValues {
+
+  val supervisingCustomsOfficeTemplate: supervising_customs_office = mock[supervising_customs_office]
+
+  val controller = new SupervisingCustomsOfficeController(
+    authenticate = mockAuthAction,
+    journeyType = mockJourneyAction,
+    navigator = navigator,
+    exportsCacheService = mockExportsCacheService,
+    mcc = stubMessagesControllerComponents(),
+    supervisingCustomsOfficePage = supervisingCustomsOfficeTemplate
+  )
+
+  val exampleCustomsOfficeIdentifier = "A1B2C3D4"
+  val exampleWarehouseIdentificationNumber = "SecretStash"
+
+  private val standardCacheModel = aDeclaration(
+    withType(DeclarationType.STANDARD),
+    withWarehouseIdentification(Some(exampleCustomsOfficeIdentifier), None, Some(exampleWarehouseIdentificationNumber), None)
+  )
+
+  private val supplementaryCacheModel = aDeclaration(
+    withType(DeclarationType.SUPPLEMENTARY),
+    withWarehouseIdentification(Some(exampleCustomsOfficeIdentifier), None, Some(exampleWarehouseIdentificationNumber), None)
+  )
+
+  private val simplifiedCacheModel = aDeclaration(
+    withType(DeclarationType.SIMPLIFIED),
+    withWarehouseIdentification(Some(exampleCustomsOfficeIdentifier), None, Some(exampleWarehouseIdentificationNumber), None)
+  )
+
+  override protected def beforeEach(): Unit = {
+    super.beforeEach()
+    authorizedUser()
+    when(supervisingCustomsOfficeTemplate.apply(any(), any())(any(), any())).thenReturn(HtmlFormat.empty)
+  }
+
+  override protected def afterEach(): Unit = {
+    Mockito.reset(supervisingCustomsOfficeTemplate)
+    super.afterEach()
+  }
+
+  "Supervising Customs Office Controller on GET request" should {
+
+    "return 200 OK" in {
+      withNewCaching(supplementaryCacheModel)
+
+      val response = controller.displayPage(Mode.Normal).apply(getRequest())
+
+      status(response) must be(OK)
+    }
+
+    "read item from cache and display it" in {
+      withNewCaching(supplementaryCacheModel)
+
+      val result = controller.displayPage(Mode.Normal).apply(getRequest())
+      await(result)
```

Comment:
no need for these to be on separate lines, await returns the content of the finished future so it's better to do

`val result = await(controller.whatever)`

also, no need to explicitly write `.apply` just do `controller.displayPage(Mode.Normal)(getRequest())`

Reply:
This applies to all tests in this class

Reply:
This has been changed